docs/developer/release-cycle.md

```diff
@@ -0,0 +1,98 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`) is released.
+
+**Note**: For the `v1.3` and `v1.4` releases we assume each release cycle will last approximately 4 months (~ 17 weeks).
+The release cycle duration will be revisited after the `v1.4.0` release.
+
+A release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * Alpha releases will be released based on the main branch only if necessary (to be determined by the release lead)
+* Week 13-14: Beta
+    * Weekly beta releases will be released based on the main branch
+        * The beta releases are meant as a preview of the upcoming release
+        * Providers can start adopting the new release based on the beta releases
+    * All changes that impact providers' adoption of the new release should be announced in the provider updates section 
+      of the office hours meeting notes and approved in the PR or issue by both approvers and key affected providers.
+* Week 15-16: RC
+    * The release branch is created
+    * Weekly RC releases will be released based on the release branch
+        * RC releases are as close as possible to the final release but they are still undergoing further testing
+    * Development of the next release can technically start on the main branch, but some changes might be delayed
+      to ensure easier cherry-picking of other changes to the release branch.
+    * There is a feature freeze on the release branch
+        * Only cherry-picks for documentation, bug fixes, test fixes and test additions are allowed
+* Week 17: GA
+    * `.0` GA release is created based on the release branch
+* After that:
+    * `.1+`: Monthly patch releases will be released based on the release branch
```

Comment:
should we make it clear that this is the responsibility of the next release team?

Reply:
Yes